File: Tours/src/main/java/com/tourism/tours/entity/KeyPoint.java
package com.tourism.tours.entity;

import com.tourism.tours.enums.KeyPointType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "key_points")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class KeyPoint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name="tour_id")
    private Tour tour;

    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    private KeyPointType type;

    private String imageUrl;

    private double latitude;

    private double longitude;
}

File: Frontend/src/types/keypoint.ts
export interface Keypoint {
    id: number;
    tourId: number;
    name: string;
    description: string;
    latitude: number;
    longitude: number;
    imageUrl?: string;
    order: number;
}

export type CreateKeypointPayload = {
    tourId: number;
    name: string;
    description: string;
    latitude: number;
    longitude: number;
    order: number;
    image?: File;
};

export type UpdateKeypointPayload = Partial<Omit<CreateKeypointPayload, 'tourId'>>;

export interface Tour {
    id: number;
    name: string;
    description: string;
    authorId: number;
    keypoints: Keypoint[];
}